x-pack/legacy/plugins/ml/public/application/services/ml_api_service/index.d.ts

```diff
@@ -124,6 +124,12 @@ declare interface Ml {
 
   results: {
     getMaxAnomalyScore: (jobIds: string[], earliestMs: number, latestMs: number) => Promise<any>;
+    fetchPartitionFieldsValues: (
+      searchTerm: Record<string, string>,
+      criteriaFields: Array<{ fieldName: string; fieldValue: any }>,
+      earliestMs: number,
+      latestMs: number
+    ) => Observable<Record<string, string[]>>;
```

Comment:
Nit: In the past we've avoided using `Record<string, ...>` and used something like `{[index: string]: string}`. For that we also have the `Dictionary` type in `common/types` to wrap that as `Dictionary<string>`. Here's a more detailed discussion on Record vs. index signatures: https://stackoverflow.com/questions/54100025/difference-between-index-signature-and-record-for-empty-object - for consistency we might stick to our `Dictionary` type and use Record if the key is a union type with distinctive values.

Reply:
Fixed in https://github.com/elastic/kibana/pull/53879/commits/c66af6e20807829e3fa8fcbb511d75cbb278d0dd